WalletWasabi/Extensions/NBitcoinExtensions.cs

```diff
@@ -353,6 +353,19 @@ void Walk(TransactionDependencyNode node)
 		}
 	}
 
+	public static ScriptType? GetScriptType(this Script script)
+	{
+		foreach (ScriptType scriptType in new ScriptType[] { ScriptType.P2WPKH, ScriptType.Taproot })
+		{
+			if (script.IsScriptType(scriptType))
+			{
+				return scriptType;
+			}
+		}
+
+		return null;
+	}
```

Comment:
You can ignore this. I would suggest to avoid returning null. The following pattern is quite common in c# (it's not a big improvement btw 

```suggestion
	public static bool TryGetScriptType(this Script script, [NotNullWhen(true)] out ScriptType? scriptType)
	{
		scriptType = new [] { ScriptType.P2WPKH, ScriptType.Taproot }.FirstOrDefault(st => script.IsScriptType(st));
		return scriptType is { };
	}
```

Reply:
This is indeed not a big improvement but I have no problem with that.

Another option is to throw an exception instead of returning `null`. This is exactly what happens in the caller methods (`EstimateInputVsize`, `OwnershipProof.VerifyOwnership()` and `Bip322.Verify()`). What do you think is better?

Reply:
Time ago I would have preferred the exception but now, after suffering them for so long, I would prefer `bool TryXXX`, `(bool ucces, T value) GetXXX` or `Result<T> GetXXX` even when none of them have a way to force the caller to check returned value.

In other words, it is up to you. I prefer my proposal but it is just a suggestion and I am not asking you to change it.

Reply:
> even when none of them have a way to force the caller to check returned value.

I suppose this is why exceptions exist.

It seems to me that using `TryGetScriptType` instead of `GetScriptType` just moves the problem to the caller methods. That means I would have to refactor them to use the `bool TryGetValue(bool succes, T value)` pattern, too.

If you are ok with that, I would leave it as it is.